lib/java/src/org/apache/thrift/transport/AutoExpandingBufferWriteTransport.java

```diff
@@ -25,10 +25,20 @@
 
   private final AutoExpandingBuffer buf;
   private int pos;
+  private int res;
 
-  public AutoExpandingBufferWriteTransport(int initialCapacity) {
+  /**
+   * Constructor.
+   * @param initialCapacity the initial capacity of the buffer
+   * @param frontReserve space, if any, to reserve at the beginning such
+   *                     that the first write is after this reserve.
+   *                     This allows framed transport to reserve space
+   *                     for the frame buffer length.
+   */
+  public AutoExpandingBufferWriteTransport(int initialCapacity, int frontReserve) {
```

Comment:
Probably assert that `initialCapacity >= frontReserve`?

Other than that, I like the fact that there's no conditional check on each `write()`